temporal/api/sdk/v1/enhanced_stack_trace.proto

```
@@ -0,0 +1,78 @@
+option ruby_package = "Temporalio::Api::Sdk::V1";
+option csharp_namespace = "Temporalio.Api.Sdk.V1";
+
+// Internal structure used to create worker stack traces with references to code.
+message EnhancedStackTrace {
+    string sdk_name = 1;
+    string sdk_version = 2;
+    map<string, FileSlice> sources = 3;
+    repeated StackTrace stacks = 4;
+}
+
+// "Slice" of a file starting at line_offset -- a line offset and code fragment corresponding to the worker's stack.
+message FileSlice {
+    // (-- api-linter: core::0141::forbidden-types=disabled
+    //     aip.dev/not-precedent: These really shouldn't have negative values. --)
+    // Only used possible to trim the file without breaking syntax highlighting.
+    uint32 line_offset = 1;
+
+    // Slice of a file with `\n` (newline) line terminator.
+    string content = 2;
+}
+
+// More specific location details of a file: its path, precise line and column numbers if applicable, and function name if available.
+// In essence, a pointer to a location in a file
+message FileLocation {
+    // Path to source file (absolute or relative).
+    // If the paths are relative, ensure that they are all relative to the same root.
+    string file_path = 1;
+    // (-- api-linter: core::0141::forbidden-types=disabled
+    //     aip.dev/not-precedent: These really shouldn't have negative values. --)
+    // Optional; If possible, SDK should send this -- this is required for displaying the code location.
+    uint32 line = 2;
+    // (-- api-linter: core::0141::forbidden-types=disabled
+    //     aip.dev/not-precedent: These really shouldn't have negative values. --)
+    // Optional; if possible, SDK should send this.
+    uint32 column = 3;
+    // Function name this line belongs to, if applicable.
+    // Used for falling back to stack trace view.
+    string function_name = 4;
+    // Flag to communicate whether a location should be hidden by default in the stack view.
+    bool hide_default = 5;
```

The SDK can simply omit this if it wants to hide the stack. Also the name `hide_default` isn't meaningful to me, there's no "default" to hide.

---

The SDK shouldn't omit it. The user might want to view something that makes sense to hide by default, but we don't want to eliminate the information entirely. Hence making it a display/UI concern rather than filtering it out send side.

It's just like the parts of a stack that are collapsed by default in IDE debugging.

---

I would just consider renaming this field to `hide_by_default` or `internal_code`.

---

I like `internal_code`; the linter doesn't like the name `hide_by_default` which is what the field was initially.